### PR TITLE
Updated to allow passing in maximum number of attempts and poll timeout

### PR DESCRIPTION
**Pull request by gigaSproule**
Attempt at fixing issue #71

### CODE REVIEW

File: embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import net.manub.embeddedkafka.ConsumerExtensions._
+import org.apache.kafka.clients.consumer.{ConsumerRecord, ConsumerRecords, KafkaConsumer}
+import org.apache.kafka.common.TopicPartition
+import org.mockito.Mockito.{times, verify, when}
+import org.scalatest.mockito.MockitoSugar
+
+import scala.collection.JavaConverters._
+
+class ConsumerOpsTest extends EmbeddedKafkaSpecSupport with MockitoSugar {
```
Comment (manub) [inline embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala]: Could you please rename this to `ConsumerOpsSpec`?

File: embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  "ConsumeLazily " should {
+    "retry to get messages with the configured maximum number of attempts when poll fails" in {
+      val consumer: KafkaConsumer[String, String] = mock[KafkaConsumer[String, String]]
+      val consumerRecords = new ConsumerRecords[String, String](mapAsJavaMap(Map.empty))
+
+      when(consumer.poll(1)).thenReturn(consumerRecords)
```
Comment (manub) [inline embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala]: maybe extracting `1` with giving a meaningful value name would be better (here and in subsequent places)

File: embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+      val maximumAttempts = 2
+      consumer.consumeLazily("topic", maximumAttempts, 1)
+
+      verify(consumer, times(maximumAttempts)).poll(1)
+    }
+
+    "not retry to get messages with the configured maximum number of attempts when poll succeeds" in {
+      val consumer: KafkaConsumer[String, String] = mock[KafkaConsumer[String, String]]
```
Comment (manub) [inline embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala]: explicit types for in-test values are not necessarily needed

File: embedded-kafka/src/main/scala/net/manub/embeddedkafka/ConsumerExtensions.scala [hunk 4]
```diff
@@ -34,18 +36,20 @@ object ConsumerExtensions {
...
       * @param topic the topic to consume
+      * @param poll  the amount of time, in milliseconds, to wait in the buffer for any messages to be available
       * @return the next batch of messages
       */
-    def getNextBatch(topic: String): Seq[(K, V)] =
+    private def getNextBatch(topic: String, poll: Long): Seq[(K, V)] =
       Try {
-        import scala.collection.JavaConversions._
-        consumer.subscribe(List(topic))
+        import scala.collection.JavaConverters._
```
Comment (manub) [inline embedded-kafka/src/main/scala/net/manub/embeddedkafka/ConsumerExtensions.scala]: Thanks for this :+1:

File: build.sbt [hunk 5]
```diff
@@ -69,6 +69,7 @@ lazy val embeddedKafka = (project in file("embedded-kafka"))
   .settings(publishSettings: _*)
   .settings(commonSettings: _*)
   .settings(commonLibrarySettings)
+  .settings(libraryDependencies += "org.mockito" % "mockito-core" % "1.8.5" % Test)
```
Comment (manub) [inline build.sbt]: Can we use a newer version of Mockito? `1.8.5` is quite old.

File: embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala (hunk 1 above)
Comment (gigaSproule) [inline embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala]: Makes sense

File: embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala (hunk 2 above)
Comment (gigaSproule) [inline embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala]: Changed

File: embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala (hunk 3 above)
Comment (gigaSproule) [inline embedded-kafka/src/test/java/net/manub/embeddedkafka/ConsumerOpsTest.scala]: Yes, sorry, was having difficulties with mocking (was trying ScalaMock, but it wasn't having it)

File: build.sbt (hunk 5 above)
Comment (gigaSproule) [inline build.sbt]: Wow, yes, it really is!

**Comment by gigaSproule** [comment]
As also mentioned in the issue, I'll fix the tests for Scala 2.11

**Comment by gigaSproule** [comment]
seems like `mapAsJavaMap` was moved in 2.12, so rather than using the deprecated `JavaConversions` object, I've use `asJava`

**Comment by manub** [comment]
Thanks @gigaSproule, will release in the next days.

**Comment by oliverlockwood** [comment]
Hi @manub, is there any chance of a release including this change, please?  It would be useful to me as well.

**Comment by manub** [comment]
Sorry, this completely slipped off my radar. Aiming to do this by the weekend.

**Comment by manub** [comment]
Released 0.13.0!

**Comment by oliverlockwood** [comment]
Nice one @manub.  I've updated my dependencies and this is working.

Apologies if this is deliberate on your part, but I noticed that https://github.com/manub/scalatest-embedded-kafka/releases has not been updated to mention the new release, and wondered if you had an issue with your automated release tagging.  Hope that's useful to know!

**Comment by manub** [comment]
Actually that should have worked - I think something went wrong with the release process and I'm not aware of it. Thanks for pointing it out!

**Pull request merged**